contracts/UniversalV3Staker.sol

```diff
@@ -355,9 +371,21 @@ contract UniversalV3Staker is IUniversalV3Staker, Multicall {
         if (calculatedRewards == 0) return;
 
         rewardUpdatedAt = timestamp;
+        // TODO: math check
+        uint24 tickBeforeUpdate = uint24(lastTick - TickMath.MIN_TICK + 1);
         lastTick = tick;
 
-        // TODO: calculate eligible rewards based on liquidity
+        uint208 liquidityLower = _cumulativeLiquidityLower.get(cfNbits, tickBeforeUpdate);
+        uint208 liquidityUpper = _cumulativeLiquidityUpper.get(cfNbits, tickBeforeUpdate);
+        uint208 liquidity = liquidityLower - liquidityUpper;
+        require(liquidity <= liquidityLower, 'UniswapV3Staker::updatePrice: overflow');
+        if (liquidity == 0) return;
+
+        uint256 rewardShare = calculatedRewards / uint256(liquidity);
```

Comment:
We may need to times uint128_max so that rewardShare is not underflowed?

Reply:
if so, will we have problems when casting it to `uint208`?

Reply:
Perhaps times uint64_max is enough - this means that the reward function can be 208 - 64 bits, which has 2 ** 144 space.

Reply:
done